metaflow/plugins/argo/argo_events_decorator.py

```diff
@@ -0,0 +1,154 @@
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+# TODO: This decorator interface can be a top-level decorator to support similar
+#       implementations for Netflix Maestro, AWS Step Functions, Apache Airflow...
+class ArgoEventsDecorator(FlowDecorator):
```

Comment:
Name should be something like `TriggerOnDecorator`. Would also suggest the name `trigger_on` as a decorator name instead of just `trigger`. This would also be consistent with the `TriggerOnFinishDecorator` below (ditto with the name).

Reply:
I kept it `@trigger` to keep it consistent with `@schedule`. Moving this file and changing class names is trivial to do and doesn't block this PR.